src/main/java/sirius/db/mixing/query/QueryCompiler.java

```diff
@@ -789,4 +792,22 @@ private C parseTag() {
 
         return null;
     }
+
+    /**
+     * Returns the optimized value for the given field respecting the Sirius DB casing annotations.
+     *
+     * @param field the field to query for
+     * @param value the value to query
+     * @return the optimized value to use in a constraint if any, or an Optional.empty() if no optimization is possible
+     */
+    protected Optional<String> getCaseOptimizedValue(Mapping field, String value) {
+        Property property = resolveProperty(field.toString()).getSecond();
+        if (property.isAnnotationPresent(LowerCase.class)) {
+            return Optional.of(value.toLowerCase());
+        }
+        if (property.isAnnotationPresent(UpperCase.class)) {
+            return Optional.of(value.toUpperCase());
+        }
+        return Optional.empty();
```

Comment:
**Optional:** Why not use a plain String as return type and give back the original value in this case?

Reply:
Using the optional I can easily decide if an optimization had occured. Then I can react on it accordingly, e.g. if value was lowercased, a case-insensitive query is not required. I made some quick and not reliable measuments using sql console with lower(value) vs value, and this can bring up to 10% performance gain.